client/payment-details/payment-method/grabpay/index.tsx

```diff
@@ -0,0 +1,111 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import React from 'react';
+
+/**
+ * Internal dependencies.
+ */
+import Detail from '../detail';
+import { PaymentMethodDetails } from 'wcpay/payment-details/types';
+import { Charge } from 'wcpay/types/charges';
+
+interface GrabPayPaymentMethodDetails extends PaymentMethodDetails {
+	grabPayTransactionId: string;
+}
+
+/**
+ * Extracts and formats payment method details from a charge.
+ *
+ * @param {Charge} charge The charge object.
+ * @return {GrabPayPaymentMethodDetails} A flat hash of all necessary values.
+ */
+const formatPaymentMethodDetails = (
+	charge: Charge
+): GrabPayPaymentMethodDetails => {
+	const { billing_details: billingDetails, payment_method: id } = charge;
+	const { name, email, formatted_address: formattedAddress } = billingDetails;
+
+	const grabPayTransactionId =
+		charge.payment_method_details.grabpay?.grabpay_id;
+
+	return {
+		id,
+		grabPayTransactionId,
+		name,
+		email,
+		formattedAddress,
+	};
+};
+
+interface GrabPayDetailsProps {
+	charge: Charge;
+	isLoading: boolean;
+}
+
+const GrabPayDetails: React.FC< GrabPayDetailsProps > = ( {
+	charge,
+	isLoading,
+} ) => {
+	const {
+		id,
+		grabPayTransactionId,
+		name,
+		email,
+		formattedAddress,
+	} = formatPaymentMethodDetails( charge );
```

Comment:
Other payment method details use placeholders that will be displayed while the remote data is loading; however, I noticed that the charge will always be fully loaded by the time this component render, according to the implementation of `PaymentDetailsPaymentMethod` in `client/payment-details/payment-method/index.js`
```
if (
		! charge.payment_method_details ||
		! charge.payment_method_details.type
	) {
		// Gracefully degrade for malformed charge objects
		return null;
	}
```